README.md

```diff
@@ -99,42 +101,46 @@ There are three ways to configure, in descending order of precedence:
 In your application, e.g.
 
     DockerClientConfig config = DockerClientConfig.createDefaultConfigBuilder()
-        .withVersion("1.16")
-        .withUri("https://my-docker-host.tld:2376")
-        .withUsername("dockeruser")
-        .withPassword("ilovedocker")
-        .withEmail("dockeruser@github.com")
-        .withServerAddress("https://index.docker.io/v1/")
-        .withDockerCertPath("/home/user/.docker")
+        .withDockerHost("tcp://my-docker-host.tld:2376")
+        .withDockerTlsVerify("1")
+        .withDockerCertPath("/home/user/.docker/certs")
+        .withDockerConfig("/home/user/.docker")
+        .withApiVersion("1.21")
+        .withRegistryUrl("https://index.docker.io/v1/")
+        .withRegistryUsername("dockeruser")
+        .withRegistryPassword("ilovedocker")
+        .withRegistryEmail("dockeruser@github.com")
```

Comment:
I think credentials in config originated from times when only docker hub was there and no private registries existed. So currently you can think of it as a default credential. There is an open issue https://github.com/docker-java/docker-java/issues/316 that is related to a refactoring of the auth configurations also. IMHO these authenication issues are out of scope of this PR and should be fixed in a separate one.